### PR TITLE
12.0 mig base exception

### PR DESCRIPTION
**Pull request by GoodERPJeff**

### CODE REVIEW

File: base_exception/README.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (elicoidal) [inline base_exception/README.rst]: see https://github.com/OCA/server-tools/pull/1392#pullrequestreview-164502064 for remarks that can apply also here.

File: base_exception/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (elicoidal) [inline base_exception/__init__.py]: remove empty line

File: base_exception/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+{
+    'name': 'Exception Rule',
+    'version': '12.0.1.0.0',
+    'category': 'Generic Modules',
+    'summary': """
+    This module provide an abstract model to manage customizable
+    exceptions to be applied on different models (sale order, invoice, ...)""",
+    'author':
+        "Akretion, Sodexis, Camptocamp, Odoo Community Association (OCA)",
+    'website': 'http://www.akretion.com',
```
Comment (elicoidal) [inline base_exception/__manifest__.py]: https://github.com/OCA/server-tools

File: base_exception/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    'depends': ['base_setup'],
```
Comment (elicoidal) [inline base_exception/__manifest__.py:16]: add image key with screenshot cover to improve the apps store visibility

File: base_exception/models/__init__.py [hunk 5]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (elicoidal) [inline base_exception/models/__init__.py]: remove empty line

File: base_exception/models/base_exception.py [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+    return wrapper
+
+
+class ExceptionRule(models.Model):
+    _name = 'exception.rule'
+    _description = "Exception Rules"
+    _order = 'active desc, sequence asc'
+
+    name = fields.Char('Exception Name', required=True, translate=True)
+    description = fields.Text('Description', translate=True)
```
Comment (elicoidal) [inline base_exception/models/base_exception.py:35]: technically string is not necessary anymore when same as column name

File: base_exception/security/base_exception_security.xml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (elicoidal) [inline base_exception/security/base_exception_security.xml:2]: add copyright

File: base_exception/tests/__init__.py [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (elicoidal) [inline base_exception/tests/__init__.py]: remove empty line

File: base_exception/views/base_exception_view.xml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    <record id="action_exception_rule_tree" model="ir.actions.act_window">
+              <field name="name">Exception Rules</field>
+              <field name="res_model">exception.rule</field>
+              <field name="view_type">form</field>
+              <field name="view_mode">tree,form</field>
+              <field name="view_id" ref="view_exception_rule_tree"/>
+              <field name="context">{'active_test': False}</field>
+          </record>
+
+    <menuitem action="action_exception_rule_tree" id="menu_action_exception" parent="base.menu_custom" sequence="90" />
```
Comment (elicoidal) [inline base_exception/views/base_exception_view.xml]: improve readability with attribute per line + add copyright

File: base_exception/wizard/base_exception_confirm_view.xml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+                            <field name="description"/>
+                        </tree>
+                    </field>
+                    <newline/>
+                </group>
+                <group>
+                    <field name="ignore" groups='base_exception.group_exception_rule_manager'/>
+                </group>
+                <footer>
+                    <button name="action_confirm" string="_Close"
```
Comment (elicoidal) [inline base_exception/wizard/base_exception_confirm_view.xml:22]: improve format + copyright

**Review by elicoidal — CHANGES_REQUESTED** [review]
LGTM.
Some details
Good job!

**Comment by elicoidal** [comment]
not sure why runbot is failing cc @pedrobaeza

**Comment by pedrobaeza** [comment]
Runbot has 2 warnings:

```
2018-10-14 13:10:53,772 150 WARNING openerp_test odoo.models: The model exception.rule.confirm has no _description
2018-10-14 13:10:53,772 150 WARNING openerp_test odoo.models: The model base.exception has no _description
```

**Comment by LoisRForgeFlow** [comment]
Commit history need some :heart:

**Comment by pedrobaeza** [comment]
Please take into account the PR #1231 for forward-porting here.

File: base_exception/models/base_exception.py (hunk 6 above)
Comment (GoodERPJeff) [reply base_exception/models/base_exception.py:35]: not agree :)

File: base_exception/security/base_exception_security.xml (hunk 7 above)
Comment (GoodERPJeff) [reply base_exception/security/base_exception_security.xml:2]: ref me some xml source link to follow

Comment (angelmoya) [reply base_exception/security/base_exception_security.xml:2]: copyright is not needed, but you can view an example on this link
https://github.com/OCA/server-tools/blob/12.0/module_auto_update/views/ir_module_module.xml#L4

**Comment by angelmoya** [comment]
@GoodERPJeff ,  will you work on this?

**Comment by pedrobaeza** [comment]
Superseeded by #1557

**Pull request closed without merging**